xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java

```diff
@@ -136,15 +144,26 @@ private int numberOfDecimals(String value) {
     }
   }
 
-  public void clearDeltaServerTime() {
-    deltaServerTime = null;
-  }
-
   public long deltaServerTime() throws IOException {
-    if (deltaServerTime == null) {
+    
+    if (deltaServerTime == null || deltaServerTimeExpire <= System.currentTimeMillis()) {
+      
+      // Do a little warm up
       Binance binance = RestProxyFactory.createProxy(Binance.class, getExchangeSpecification().getSslUri());
-      deltaServerTime = binance.time().getServerTime().getTime() - System.currentTimeMillis();
+      Date serverTime = new Date(binance.time().getServerTime().getTime());
+      serverTime = new Date(binance.time().getServerTime().getTime());
```

Comment:
why call this twice in a row?

Reply:
Because this is also likely going to be the very first call that initializes the HTTPS connection. By the time I get a response from that fist call it may not reflect the actual server time any more. The second call is likely to have less latency, so the time delta is more accurate.

This procedure happens after every 10mins